cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java

```diff
@@ -53,11 +57,14 @@ public Map<String, VirtualMachine> getVirtualMachinesByName(AzureClient azureCli
         }
         if (hasNoVmInResourceGroup(virtualMachines)) {
             LOGGER.info("We could not receive any VM in resource group. Let's try to fetch VMs by instance ids.");
+            int successfullyFetchedInstanceCount = 0;
             for (String privateInstanceId : privateInstanceIds) {
                 VirtualMachine virtualMachineByResourceGroup = azureClient.getVirtualMachineByResourceGroup(resourceGroup, privateInstanceId);
                 if (virtualMachineByResourceGroup == null) {
                     LOGGER.info("Could not find vm with private id: " + privateInstanceId);
                 } else {
+                    successfullyFetchedInstanceCount++;
+                    errorIfInstancesActuallyExistOnAzure(privateInstanceIds.size(), successfullyFetchedInstanceCount);
```

Comment:
If I understand correctly, from now on the logic will be:
1. check vms if there are in resourceGroup
2. if we find the MAXIMUM_INSTANCE_COUNT_FOR_ONE_BY_ONE_FETCHING number of machines we throw an error.


Where I believe the logic should be about:
if we find 5 or more instance we assume all is there and we continue with the operation. 
This would be only true if we only use the result of this method for validation.

Reply:
We throw an error, if the number of instances found during one-by-one fetching is larger than `SUCCESSFUL_INSTANCE_FETCH_COUNT_BEFORE_RETRY` AND the total number of instances we are looking for is larger than `MAXIMUM_INSTANCE_COUNT_FOR_ONE_BY_ONE_FETCHING`.

This way, the whole operation will be retried if there are a lot of instances to be fetched and we know at least some of them are there.

This solution only prevents us making potentially 10s or 100s of calls when fetching one-by-one, by forcing the list call to be retried.

Should we just return a partial result, if we know some of the instances are there, instead?